src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -609,7 +625,10 @@ class FastSync(
 
     def assignWork(peer: Peer): Unit = {
       if (syncState.bestBlockHeaderNumber < syncState.safeDownloadTarget || syncState.blockChainWorkQueued) {
-        assignBlockchainWork(peer)
+        if((syncState.pendingMptNodes.nonEmpty || syncState.pendingNonMptNodes.nonEmpty) && math.random < 0.05)
```

Comment:
this `syncState.pendingMptNodes.nonEmpty || syncState.pendingNonMptNodes.nonEmpty` could be a method in syncState

Reply:
O am not sure if it is needed to move this method. It is used only once.

Reply:
But it will name a concept behind some chunk of code which makes reading and understanding code much easier.